sdks/java/core/src/test/java/com/google/cloud/dataflow/sdk/io/AvroIOTest.java

```diff
@@ -222,6 +222,37 @@ public void testAvroSinkWrite() throws Exception {
     }
   }
 
-  // TODO: for Write only, test withSuffix, withNumShards,
+  @SuppressWarnings("deprecation") // using AvroCoder#createDatumReader for tests.
+  @Test
+  public void testAvroSinkShardedWrite() throws Exception {
```

Comment:
Agreed; this would be best done by using the PipelineOptions.getTempLocation() for the location of the temporary files, and IOChannelUtils to either create input files or write to output files.

Filed [BEAM-181](https://issues.apache.org/jira/browse/BEAM-181)